### ISSUE TITLE
How to use EasyRefresh

### ISSUE THREAD

**Issue by zhongcg** [rvamsikrishna/inview_notifier_list#14]

**Comment by zhongcg** [comment]
@override
  Widget build(BuildContext context) {
    return NotificationListener(
        onNotification: (notification) { 
          if (notification is ScrollEndNotification) {
            isTouch = false;
            setState(() {});
          }
          if (notification is ScrollStartNotification) {
            isTouch = true;
          }
          return false;
        },
        child: LoadStateLayout(
            state: _loadState,
            errorRetry: _onRefresh,
            emptyRetry: _onRefresh,
            successWidget: EasyRefresh(
                onRefresh: _onRefresh,
                onLoad: _loadMore,
                child: getListView(context))));
  }
  
  getListView(BuildContext context) {
    return InViewNotifierCustomScrollView(
        shrinkWrap: true,
        initialInViewIds: ['0'],
        isInViewPortCondition:
            (double deltaTop, double deltaBottom, double viewPortDimension) {
          return deltaTop < (0.5 * viewPortDimension) &&
              deltaBottom > (0.5 * viewPortDimension);
        },
        controller: _scrollController,
        slivers: <Widget>[
          SliverList(
            delegate: SliverChildBuilderDelegate(
              (context, index) {
                GlobalKey bottomKey = GlobalKey();
                return InViewNotifierWidget(
                  id: '$index',
                  builder: (BuildContext context, bool isInView, Widget child) {
                    final String inViewTxt = isInView ? 'inView' : 'notInView';
                    print('$inViewTxt $index');
                    return buildItems(
                              videolist[index], bottomKey, isInView);
                  },
                );
              },
              childCount: videolist.length,
            ),
          )
        ]);

**Comment by zhongcg** [comment]
can not change isInView Index . 
isInView index still 0

**Comment by rvamsikrishna** [comment]
I could not understand your question completely

**Comment by sdegenaar** [comment]
Hi, I can confirm that EasyRefresh is working if you enable in InViewNotifierList
 shrinkWrap: true,
 physics: ScrollPhysics(),

but the isInView does not change

**Comment by zhujiahong** [comment]
> Hi, I can confirm that EasyRefresh is working if you enable in InViewNotifierList
> shrinkWrap: true,
> physics: ScrollPhysics(),
> 
> but the isInView does not change

I ran into the same problem ，when I add  
shrinkWrap: true,
physics: ScrollPhysics(),
isInView can not working

**Comment by sdegenaar75** [comment]
yep be nice to get this working. I have tried lots of things. Let me know if you have any success or the author has any ideas. Thanks

**Comment by sdegenaar75** [comment]
Hi, any news on this or suggestions? Thanks!

**Comment by vamsi-kleargroup** [comment]
I've added a `scrollViewWrapper` parameter to `InViewNotifierList` and `InViewNotifierCustomScrollView` that lets you wrap the internal scroll view with any refresh widget — no need to access `InViewNotifier` directly.

Example usage:
```dart
InViewNotifierList(
  scrollViewWrapper: (scrollView) => RefreshIndicator(
    onRefresh: _onRefresh,
    child: scrollView,
  ),
  // ...
)
```

This also works with third-party libraries like `easy_refresh`:
```dart
InViewNotifierList(
  scrollViewWrapper: (scrollView) => EasyRefresh(
    onRefresh: () async => _onRefresh(),
    child: scrollView,
  ),
  // ...
)
```

The feature is on the [`feat/scroll-view-wrapper`](https://github.com/rvamsikrishna/inview_notifier_list/tree/feat/scroll-view-wrapper) branch. If this approach works for your use case, let me know and I'll merge it into the next release.

**Comment by rvamsikrishna** [comment]
Update: this is now fixed in **v4.1.0** 🎉

The new `scrollViewWrapper` callback (on both `InViewNotifierList` and `InViewNotifierCustomScrollView`) is built for exactly this thread. It wraps the notifier's **internal** scroll view, so `EasyRefresh` (or `RefreshIndicator`, `SmartRefresher`, …) works *and* in-view detection keeps updating:

```dart
InViewNotifierCustomScrollView(
  scrollViewWrapper: (scrollView) => EasyRefresh(
    onRefresh: _onRefresh,
    onLoad: _loadMore,
    child: scrollView,
  ),
  isInViewPortCondition: (deltaTop, deltaBottom, vp) =>
      deltaTop < 0.5 * vp && deltaBottom > 0.5 * vp,
  slivers: <Widget>[ /* ... */ ],
)
```

Why `isInView` used to get stuck: wrapping the whole widget in `EasyRefresh` (with the `shrinkWrap: true` + `physics: ScrollPhysics()` workaround) moved the actual scrolling up to the outer widget, so the notifier's internal scroll view never scrolled and never emitted the scroll notifications detection relies on. `scrollViewWrapper` keeps the notifier's own scroll view as the one that scrolls, so notifications still reach it — no more workaround needed.

There's also a new pull-to-refresh tab in the example app demonstrating both `EasyRefresh` and `RefreshIndicator`.

Available now: https://pub.dev/packages/inview_notifier_list/versions/4.1.0

Thanks @zhongcg for the original report, and @sdegenaar @zhujiahong @sdegenaar75 for the extra detail and patience. Closing as resolved — please reopen if anything still misbehaves on 4.1.0. 🙏
